fusor - updates to support initial rpm build of fusor_ui

bbuckingham opened the pull request:
This PR contains the initial changes to support building an RPM for rubygem-fusor_ui. 

Tested this out by performing scratch builds using the koji.katello.org.

There is at least one issue that I am aware of with respect to compilation of assets which currently results in errors when attempting to access the new UI navigation items (e.g. Deployments & New Deployment). Need to investigate how assets should be correctly recompiled from a foreman plugin.

The error looks similar to : 
```
Error compiling asset fusor-demo.js:
LoadError: cannot load such file -- uglifier
  (in /opt/rh/ruby193/root/usr/share/gems/gems/fusor_ui-0.0.1/app/assets/javascripts/fusor-demo.js)
```

pitr-ch commented:
@bbuckingham any update on this PR? So far I did not review since you mentioned there is an error.

bbuckingham commented:
@pitr-ch, With the changes in this PR, we are able to build an RPM of fusor_ui.  The last 2 commits support addressing issues with asset compilation in koji.  Those changes need to be handled differently since the source is a result of the 'ember build' process performed in the fusor-demo repository and then copied to the fusor repository.  Unfortunately, we cannot simply update the files in the fusor-demo repository as vendor files appear to be retrieved externally as part of the process (and are not built from content in our git repo).

If possible, I'd recommend that we move forward with this PR and allow @isratrade to work through the proper workflow to support rpm production builds with emberjs/embercli.  I suspect that it will involve downloading and storing the vendor assets in git and having the ember build process utilize the files in git versus retrieving them from elsewhere.  From feedback from the folks that had to do the same for angular/bastion, this is also going to be required in order to support downstream builds.

pitr-ch commented:
Sound good to me. @isratrade if ACK, please merge.

isratrade commented on ui/app/assets/stylesheets/fusor_ui/fusor-demo.css:
```diff
@@ -619,7 +619,7 @@ select{padding: initial;}
   border-color: #f5f5f5;
   background-color: #f5f5f5;
-  @include box-shadow(none);
+  box-shadow: none;
```
fixed in repo

isratrade commented on ui/app/assets/javascripts/fusor_ui/vendor-fusor.js:
```diff
@@ -78570,7 +78570,7 @@ helpers = this.merge(helpers, Ember.Handlebars.helpers); data = data || {};
  */
 
-if (typeof jQuery === 'undefined') { throw new Error('Bootstrap\'s JavaScript requires jQuery') }
+//if (typeof jQuery === 'undefined') { throw new Error('Bootstrap\'s JavaScript requires jQuery') }
```
I will try to figure out how to automate deleting this line.
